### ISSUE TITLE
Version changed to something like "0.0.1-post.120+c5d6581" every time project is opened in PyCharm

### ISSUE THREAD

**Issue by danhje** [mtkennerly/poetry-dynamic-versioning#71]
Starting a few weeks ago, simply opening any project with dynamic versioning in PyCharm causes the version number to changed in pyproject.toml and __init__.py, and `enable` is set to `false` in the config, and the changes never get reverted. This started before PyCharm added poetry-support. Any idea what is causing this?

**Comment by mtkennerly** [comment]
I don't use PyCharm, but I just did a fresh install of it, and I wasn't able to reproduce this issue. I opened a project in PyCharm, then entered the Poetry executable location when prompted.

Here's my environment:

* Windows 11
* Python 3.7.9
* PyCharm Community 2021.3.2
* poetry-dynamic-versioning 0.13.1
* Poetry 1.1.12
* Poetry installation method: `pip install poetry`

Could you share your environment? Do you have any additional PyCharm plugins that might affect this?

**Comment by danhje** [comment]
* Ubuntu 20.04.4 LTS
* Python 3.8.2 / 3.9.10 / 3.10.2 (different projects use different version, managed through pyenv, not sure if the behavior is seen with all of them)
* PyCharm Professional 2021.3
* Poetry 1.1.2
* Poetry install method: `pip install poetry`

How do I get the version of poetry-dynamic-versioning?

A few colleges have the same issue, and they have pretty much an identical environment. 

I've noticed that it doesn't always happen. I'll see if I can spot any patterns, and report back.

**Comment by mtkennerly** [comment]
You can check the plugin version in the output from `pip list`.

**Comment by mtkennerly** [comment]
I can partially reproduce this in a Linux VM.

When I initially imported the project, I saw the changes appear, and they were reverted when I closed PyCharm. Each time I reopened PyCharm after that, I saw the changes appear twice, for about half a second each time, and then they were automatically reverted while PyCharm was still running. This behavior would make sense if PyCharm is running the Poetry CLI, for example.

However, I didn't see the changes get stuck like you mentioned. I'm not sure why that's behaving differently; my best guess would be if PyCharm were running/importing Poetry and then killing a process abruptly so that the plugin couldn't clean up after itself. I also don't know why it would behave differently on Windows; that's hard to say without knowledge of PyCharm's internals.

**Comment by mtkennerly** [comment]
Did some more testing (Linux, PyCharm Community 2022.1.1, Poetry 1.1.13, poetry-dynamic-versioning 0.17.0). When I open `poetry-dynamic-versioning/tests/project` in PyCharm, there's a lot of back and forth:

```
$ tail -n 25 -F pyproject.toml | grep 'version ='
version = "0.0.999"
tail: pyproject.toml: file truncated
version = "0.17.0"
tail: pyproject.toml: file truncated
version = "0.0.999"
tail: pyproject.toml: file truncated
version = "0.17.0"
tail: pyproject.toml: file truncated
version = "0.17.0"
tail: pyproject.toml: file truncated
version = "0.0.999"
tail: pyproject.toml: file truncated
version = "0.17.0"
tail: pyproject.toml: file truncated
version = "0.0.999"
```

The exact sequence is different sometimes, but I still haven't seen it leave the dynamic version in the file.

If someone can help to define some minimum steps to reproduce the behavior and confirm that it's still an issue, then I can investigate further, but for now, I'm going to close this.

For my testing, I've used the following Vagrantfile:

```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "ubuntu/focal64"
  config.vm.network "public_network"
  config.vm.provider "virtualbox" do |vb|
    vb.gui = true
    vb.memory = 4096
  end

  config.vm.provision "shell", inline: <<-SHELL
    sudo apt-get update
    sudo apt-get install -y ubuntu-desktop python3-pip
  SHELL
end
```

I have this box locally:

```
$ vagrant box list
ubuntu/focal64 (virtualbox, 20211026.0.0)
```

In the VM, I ran:

```
startx
pip3 install poetry
pip3 install poetry-dynamic-versioning
git clone https://github.com/mtkennerly/poetry-dynamic-versioning.git
cd poetry-dynamic-versioning/tests/project
python3 -m poetry config virtualenvs.in-project true
python3 -m poetry install
```

And then I opened `~/poetry-dynamic-versioning/tests/project` in PyCharm.

**Comment by caniko** [comment]
I also have this issue.

**Comment by danhje** [comment]
I haven't had any similar problems for months, and I work with multiple projects that use poetry-dynamic-versioning in PyCharm daily. No idea what fixed it, unfortunately.

**Comment by caniko** [comment]
OK, it happened when I did `poetry update` @mtkennerly. Please re-open.

**Comment by mtkennerly** [comment]
@caniko I just tried testing this again, and I still see it revert the changes when it's done, even if I run `poetry update` while pyproject.toml is open in PyCharm :/

By any chance, do you have file substitutions configured for pyproject.toml? There was a bug fixed in v1.1.1 regarding that.
